packages/frontend/src/hooks/usePaginationAndFilter.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback } from 'react'
+import { useSearchParams } from 'react-router-dom'
+
+interface SearchParams {
+  page: number
+  input: string
+  status: string
+}
+
+export function usePaginationAndFilter() {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const page = parseInt(searchParams.get('page') || '', 10) || 1
+  const input = searchParams.get('input') || ''
+  const status = searchParams.get('status') || ''
+
+  // format search params for empty strings and first page
+  const setFormattedSearchParams = useCallback(
+    (params: Partial<SearchParams>) => {
+      setSearchParams((currentSearchParams) => {
```

Comment:
actually just curious do u need to create a new params object:
`const updatedSearchParams = new URLSearchParams(currentSearchParams);`
instead of directly mutating the object

Reply:
no difference